packages/components/Shimmer/src/Shimmer.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import { ensureNativeComponent } from '@fluentui-react-native/component-cache';
+
+const shimmerName = 'Shimmer';
+
+const NativeShimmerView = ensureNativeComponent('MSFShimmerView');
+
+/**
+ * Appearance of the views that are shimmered
+ */
+export interface ShimmerViewAppearance {
+  /**
+   * Color to tint the shimmer boxes. Defaults to the Fabric default color.
+   */
+  tintColor?: number;
+
+  /**
+   * Corner radius on each view.
+   */
+  cornerRadius?: number;
+
+  /**
+   * Corner radius on each UILabel. Set to < 0 to disable and use default `cornerRadius`.
+   */
+  labelCornerRadius?: number;
+
+  /**
+   * usesTextHeightForLabels: True to enable shimmers to auto-adjust to font height for a UILabel -- this
+   * will more accurately reflect the text in the label rect rather than using the bounding box. `labelHeight`
+   * will take precedence over this property.
+   */
+  usesTextHeightForLabels?: boolean;
+
+  /**
+   * If greater than 0, a fixed height to use for all UILabels. This will take precedence
+   * over `usesTextHeightForLabels`. Set to less than 0 to disable.
+   */
+  labelHeight?: number;
+}
+
+/**
+ * Appearance of the shimmer itself (the animation appearance)
+ */
+export interface ShimmerAppearance {
+  alpha?: number;
+  width?: number;
+
+  /**
+   * Angle of the direction of the gradient, in radian. 0 means horizontal, Pi/2 means vertical.
+   */
+  angle?: number;
+
+  /**
+   * Speed of the animation, in point/seconds.
+   */
+  speed?: number;
+
+  /**
+   * Delay between the end of a shimmering animation and the beginning of the next one, in seconds
+   */
+  delay?: number;
+}
+
+/**
+ * ViewProps props, based off of the standard react-native ViewProps with some new extensions
+ */
+export type NativeShimmerViewProps<TBase = IViewProps> = TBase & {
+  /**
+   * Appearance of the views that are shimmered. If not passed, the default will be used.
+   */
+  appearance?: ShimmerViewAppearance;
+
+  /**
+   * Appearance of the shimmer itself (the animation appearance)
+   */
+  shimmerAppearance?: ShimmerAppearance;
+};
+
+/**
+ * Shimmer Token is a flat list of all the properties we can set on the native view
+ */
+export type ShimmerToken = ShimmerAppearance & ShimmerViewAppearance;
+
+interface ShimmerType {
+  props: ViewProps;
+  slotProps: { root: NativeShimmerViewProps };
+  tokens: ShimmerToken;
+}
+
+export const Shimmer = compose<ShimmerType>({
+  displayName: shimmerName,
+  tokens: [
+    ({
+      tintColor: processColor('#F1F1F1'),
+      cornerRadius: 4,
+      labelCornerRadius: 2,
+      usesTextHeightForLabels: false,
+      labelHeight: 11,
+      alpha: 0.4,
+      width: 180,
+      angle: -(Math.PI / 45.0),
+      speed: 350,
+      delay: 0.4
+    }),
+    shimmerName,
+  ],
+  slotProps: {
+    root: buildProps<NativeShimmerViewProps, ShimmerToken>(
+      (tokens: ShimmerToken) => ({
+        appearance: {
+          tintColor: tokens.tintColor,
```

Comment:
Since you are using explicit set syntax to map to the sub-objects the names don't have to exactly match. E.g. you could rename tintColor to viewTintColor if you wanted. Just something to consider. It looks like you have good descriptions on the properties in the interface, you can just design what a flat interface would look like without being constrained to the native interface exactly.

Reply:
I ended up just naming tintColor viewTintColor, and renaming a lot of the native properties with what I feel are better names. It's a flat interface now anyway, so this should be resolved